sklearn/cluster/_hdbscan/_tree.pyx

```diff
@@ -212,55 +212,32 @@ cdef dict _compute_stability(
     cdef:
         cnp.float64_t[::1] result, births
         cnp.intp_t[:] parents = condensed_tree['parent']
-        cnp.float64_t[:] lambdas = condensed_tree['value']
-        cnp.intp_t[:] sizes = condensed_tree['cluster_size']
 
         cnp.intp_t parent, cluster_size, result_index
-        cnp.float64_t lambda_val, child_size
+        cnp.float64_t lambda_val
+        CONDENSED_t condensed_node
         cnp.float64_t[:, :] result_pre_dict
         cnp.intp_t largest_child = condensed_tree['child'].max()
         cnp.intp_t smallest_cluster = np.min(parents)
         cnp.intp_t num_clusters = np.max(parents) - smallest_cluster + 1
-        cnp.ndarray sorted_child_data = np.sort(condensed_tree[['child', 'value']], axis=0)
-        cnp.intp_t[:] sorted_children = sorted_child_data['child'].copy()
-        cnp.float64_t[:] sorted_lambdas = sorted_child_data['value'].copy()
-        cnp.intp_t child, current_child = -1
-        cnp.float64_t min_lambda = 0
 
     largest_child = max(largest_child, smallest_cluster)
     births = np.full(largest_child + 1, np.nan, dtype=np.float64)
 
-    if largest_child < smallest_cluster:
-        largest_child = smallest_cluster
-
     births = np.full(largest_child + 1, np.nan, dtype=np.float64)
-    for idx in range(condensed_tree.shape[0]):
-        child = sorted_children[idx]
-        lambda_val = sorted_lambdas[idx]
-
-        if child == current_child:
-            min_lambda = min(min_lambda, lambda_val)
```

Comment:
Note that `child != current_child` in this algorithm since each child is unique to a single parent and hence only appears once. I suspect the removal of this line also simplifies the algorithm enough to remove the need for sorting entirely.